File: Homework/quoc_khanh/lesson_1/191.Numberof1Bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n > 0:
            n &= n - 1
            count += 1
        return count

s = Solution()
print(s.hammingWeight(0) == 0)
print(s.hammingWeight(1) == 1)
print(s.hammingWeight(2) == 1)
print(s.hammingWeight(3) == 2)
print(s.hammingWeight(4) == 1)
print(s.hammingWeight(5) == 2)
print(s.hammingWeight(6) == 2)
print(s.hammingWeight(7) == 3)
print(s.hammingWeight(8) == 1)
print(s.hammingWeight(9) == 2)
print(s.hammingWeight(16) == 1)
print(s.hammingWeight(100) == 3)
print(s.hammingWeight(1234) == 5)

File: Homework/Hung_Tran/Lesson1/191_number_of_1_bits.js
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function (n) {
  var n_binary = n.toString(2);
  var n_array = n_binary.split("");
  var count = 0;
  for (let i = 0; i < n_array.length; i++) {
    if (n_array[i] == "1") {
      count++;
    }
  }
  return count;
};
